Added the ability to set the background color

PGMacDesign opened the pull request:
Added 2 methods for setting the background color of the Elastic view. It should take in either an int color or a String and parse the color. If the string is null or the parsing fails, nothing will be set and it will retain the original color

PGMacDesign committed:
Added the ability to set the background color

Added 2 methods for setting the background color of the Elastic view. It should take in either an int color or a String and parse the color. If the string is null or the parsing fails, nothing will be set and it will retain the original color

Tibolte commented on elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java:
```diff
@@ -107,5 +107,33 @@ public void onEnterAnimationFinished() {
     }
+    
+    
+    /////////////////////////////////////////////////////////////////////////////
```
You can remove that comment, it's not really necessary

Tibolte commented on elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java:
```diff
@@ -107,5 +107,33 @@ public void onEnterAnimationFinished() {
+        try {
+            int color = Color.parseColor(passedColor);
+            this.setBackgroundColor(color);
+        } catch (Exception e){}
```
[Don't catch generic exception](https://source.android.com/source/code-style.html#dont-catch-generic-exception)

Tibolte commented on elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java:
```diff
@@ -107,5 +107,33 @@ public void onEnterAnimationFinished() {
+     * Set the background color of the Elastic Download View
+     * @param passedColor int Color Color to set the background
+     */
+    public void setBackgroundColor(int passedColor){
```
Take a look at how the rest of the file is formatted (I'm talking about braces here), there's always one space between one parenthesis and any curly brace. Try to keep some consistency on the code styling :)

PGMacDesign commented:
Thanks for the tips. Were you wanting me to change the code prior to merge? Or were you just offering advice?

Tibolte commented:
That's better if you change it, it should be quick to do it ;)

PGMacDesign committed:
Fixed some grammatical errors

Adjusted text and changed catch argument. Omitted an e.printStackTrace() as I was unsure if it was desired. Let me know if you want it implemented.

PGMacDesign commented:
Updated with said fixes. Let me know if you need anything else changed on there, I will be happy to do it.

Tibolte commented on elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java:
```diff
@@ -107,5 +107,30 @@ public void onEnterAnimationFinished() {
+     *                    Color.parseColor(string)
+     */
+    public void setBackgroundColor(String passedColor) {
+        if(passedColor == null){
```
this `if` needs a space between itself and the parenthesis (like the rest of the code).

Tibolte replied:
Just missing `if (passedColor == null)` (adding a white space) and it's good to go.

Tibolte commented on elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java:
```diff
@@ -107,5 +107,30 @@ public void onEnterAnimationFinished() {
+        try {
+            int color = Color.parseColor(passedColor);
+            this.setBackgroundColor(color);
+        } catch (IllegalArgumentException e) {}
```
Better! What we usually do in the case is use `e.printstacktrace();` to notify in logcat that something when wrong, or you can use a log message with `Log.e`. Either way we should do something here, so developers can get notified when a wrong color has been passed to this function.

PGMacDesign committed:
Adding in printStackTrace

e.printStackTrace(); under catch for null Strings or bad color parsing strings sent

PGMacDesign committed:
Adding a space for formatting
